src/Core/Layer/Layer.js

```diff
@@ -175,4 +182,77 @@ const ImageryLayers = {
     },
 };
 
+/**
+ * Interface implemented by {@link GeometryLayer} instances capable of displaying color layers
+ * @interface
+ */
+function CanDisplayColorLayer() {}
+/**
+ * Add a color layer to display on this geometry
+ *
+ * @function
+ * @name CanDisplayColorLayer#addColorLayer
+ * @param {Object} colorLayer - property bag defining the layer to display.
+ * @param {string} colorLayer.id - unique identifier of this layer
+ * @param {string} colorLayer.protocol - one of wms, wmts, wmtsc, tms, static.
+ * Each protocol has a set of specific options that are documented in XXX.
+ * @param {Extent} extent - Extent covered by the layer. Note that it can be different
+ * than the extent of the underlying geometry.
+ * @param {number} colorLayer.strategy.type - Strategy to use to download elements.
+ */
+/**
+ * Remove a displayed color layer
+ *
+ * @function
+ * @name CanDisplayColorLayer#removeColorLayer
+ * @param {({Layer}|string)} - the layer to remove or its id
+ */
+
+ /**
+ * Interface implemented by {@link GeometryLayer} instances capable of displaying color layers
+ * @interface
+ */
+function CanDisplayElevationLayer() {}
+/**
+ * Use an elevation data source to deform the underlying geometry
+ *
+ * @function
+ * @name CanDisplayElevationLayer#addElevationLayer
+ * @param {Object} elevationLayer - property bag defining the layer to display.
+ * @param {string} elevationLayer.id - unique identifier of this layer
+ * @param {string} elevationLayer.protocol - one of wms, wmts, wmtsc, static.
+ * Each protocol has a set of specific options that are documented in XXX.
+ * @param {Extent} extent - Extent covered by the layer. Note that it can be different
+ * than the extent of the underlying geometry.
+ * @param {number} elevationLayer.strategy.type - Strategy to use to download elements.
+ */
+/**
+ * Remove an elevation layer
+ *
+ * @function
+ * @name CanDisplayElevationLayer#removeElevationLayer
+ * @param {({Layer}|string)} - the layer to remove or its id
+ */
+
+ /**
+ * Interface implemented by {@link GeometryLayer} instances capable of displaying a feature layers
+ * @interface
+ */
+function CanDisplayFeatureLayer() {}
+/**
+ * Display a feature layers on the underlying geometry
```

Comment:
I think we need a definition of what a feature layer is. I guess it is a layer made of data fetched from a wfs (or wfs-like) source?